src/core/middleware/store.ts

```diff
@@ -0,0 +1,54 @@
+import { destroy, invalidator, create } from '../vdom';
+import injector from '../middleware/injector';
+import Store, { StatePaths, Path } from '../../stores/Store';
+import { Process } from '../../stores/process';
+
+const factory = create({ destroy, invalidator, injector });
+
+export const createStoreMiddleware = <S = any>(initial?: (store: Store<S>) => void) => {
+	let store = new Store<S>();
+	let initialized = false;
+	initial && initial(store);
+	const storeMiddleware = factory(({ middleware: { destroy, invalidator, injector } }) => {
+		const handles: any[] = [];
+		destroy(() => {
+			let handle: any;
+			while ((handle = handles.pop())) {
+				handle();
+			}
+		});
+		if (!initialized) {
```

Comment:
Does a one time check for a store in the application registry, if it exists the middleware will use that instead